promtsdb/promtsdb.go

```diff
@@ -71,26 +85,32 @@ func (q *V3ioPromQuerier) Select(params *storage.SelectParams, oms ...*labels.Ma
 			}
 		}
 	}
-	set, err := q.q.SelectProm(name, functions, params.Step, filter, noAggr)
+	set, err := promQuery.v3ioQuerier.SelectProm(name, functions, params.Step, filter, noAggr)
 	return &V3ioPromSeriesSet{s: set}, err
 }
 
 // LabelValues returns all potential values for a label name.
-func (q *V3ioPromQuerier) LabelValues(name string) ([]string, error) {
-	return q.q.LabelValues(name)
+func (promQuery *V3ioPromQuerier) LabelValues(name string) ([]string, error) {
+	return promQuery.v3ioQuerier.LabelValues(name)
 }
 
 // Close releases the resources of the Querier.
-func (q *V3ioPromQuerier) Close() error {
+func (promQuery *V3ioPromQuerier) Close() error {
 	return nil
 }
 
-func match2filter(oms []*labels.Matcher) (string, string, string) {
-	filter := []string{}
+func match2filter(oms []*labels.Matcher, logger logger.Logger) (string, string, string) {
+	var filter []string
 	agg := ""
 	name := ""
 
+	if logger == nil {
+		// TODO: handle and return the error
+		logger, _ = utils.NewLogger(config.DefaultLogLevel)
```

Comment:
It seems to me that instead of handling the error you could just not instantiate a logger here... I couldn't find any callers that don't pass a logger into this method.

Reply:
correct, I use to pass it everywhere